backend/database/store/store.go

```diff
@@ -0,0 +1,78 @@
+package store
+
+import (
+	"context"
+	"fmt"
+	"runtime"
+	"time"
+
+	"github.com/GenerateNU/sac/backend/constants"
+	m "github.com/garrettladley/mattress"
+	"github.com/redis/go-redis/v9"
+)
+
+type StoreClientInterface interface {
+	Set(key string, value interface{}, expiration time.Duration) error
+	Get(key string) (string, error)
+	Del(key string) error
+	Exists(key string) (bool, error)
+	SAdd(key string, members ...interface{}) error
+	SIsMember(key string, member interface{}) (bool, error)
+	FlushAll() error
+	Close() error
+}
+
+type RedisClient struct {
+	Client *redis.Client
+	Ctx    context.Context
```

Comment:
what does redis client need context for? context is usually for passing into/through units of work, makes more sense as an argument to funcs of the interface rather than as shared state in the client